internal/http/middleware_test.go

```diff
@@ -0,0 +1,103 @@
+package http_test
```

Comment:
to have a dedicated test for this, what do you think if we extract the `ErrorMiddleware` as it's own package? so the tests that will be run on the project will be exclusively for the app, not the tool.

Reply:
I think the middleware itself should be considered as part of the app.
For example, this boilerplate only includes:
- 404 -> ErrNotFound
- 400 -> ConstraintError
- 408 -> slow responses
Other projects might uses 401, 403, 409, 422 and those projects should update this file.

Reply:
I still think that this can be abstracted, let me tinker some code first.

Reply:
okay, nevermind, after a few tinkering, I agree the middleware should stay